Content.Server/GameObjects/Components/VendingMachines/VendingMachineComponent.cs

```diff
@@ -129,6 +129,18 @@ private void UpdatePower(PowerChangedMessage args)
         {
             var state = args.Powered ? VendingMachineVisualState.Normal : VendingMachineVisualState.Off;
             TrySetVisualState(state);
+
+            // Pause/resume advertising if advertising component exists and not broken
+            if (!Owner.TryGetComponent(out AdvertiseComponent? advertiseComponent) || _broken) return;
+
+            if (Powered)
+            {
+                advertiseComponent.Resume();
+            }
+            else
+            {
+                advertiseComponent.Pause();
+            }
```

Comment:
This is kinda dirty and should be an event or the likes to try and separate out the functionality.

Reply:
Will be kept like this as decided on Discord (https://discord.com/channels/310555209753690112/770682801607278632/832591047888601099).